cmd/api/src/api/v2/auth/auth.go

```diff
@@ -480,6 +480,23 @@ func (s ManagementResource) CreateUser(response http.ResponseWriter, request *ht
 				api.HandleDatabaseError(request, response, err)
 			} else {
 				userTemplate.SAMLProviderID = null.Int32From(samlProvider.ID)
+				userTemplate.SSOProviderID = null.Int32From(samlProvider.SSOProviderID.Int32)
+			}
+		} else if createUserRequest.SSOProviderID != "" {
```

Comment:
I believe we can also have the `SSOProviderID` be a `null.Int32`, which allows the int to be "optional".

Then you could use `if createUserRequest.SSOProviderID.Valid` rather than checking strings.

Not marking as required, though, since we currently do this for SAML ID as well